feat(jupyter): Add Jupyter

Dyllan2000alfa opened the pull request:
**Description**
Add the official Jupyter docker images 
<!--
-->
⚒️ Fixes  # <!--(issue)-->

**⚙️ Type of change**

- [x] ⚙️ Feature/App addition
- [ ] 🪛 Bugfix
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] 🔃 Refactor of current code

**🧪 How Has This Been Tested?**
<!--
-->
Haven't tested anything yet. Just created the basics so I can create a draft PR for CI

**📃 Notes:**
<!-- Please enter any other relevant information here -->

**✔️ Checklist:**

- [ ] ⚖️ My code follows the style guidelines of this project
- [ ] 👀 I have performed a self-review of my own code
- [ ] #️⃣ I have commented my code, particularly in hard-to-understand areas
- [ ] 📄 I have made corresponding changes to the documentation
- [ ] ⚠️ My changes generate no new warnings
- [ ] 🧪 I have added tests to this description that prove my fix is effective or that my feature works
- [ ] ⬆️ I increased versions for any altered app according to semantic versioning

**➕ App addition**

If this PR is an app addition please make sure you have done the following.

- [x] 🪞 I have opened a PR on [truecharts/containers](https://github.com/truecharts/containers) adding the container to TrueCharts mirror repo.
- [x] 🖼️ I have added an icon in the Chart's root directory called `icon.png`

---

_Please don't blindly check all the boxes. Read them and only check those that apply.
Those checkboxes are there for the reviewer to see what is this all about and
the status of this PR with a quick glance._

Dyllan2000alfa committed:
Initial work for Jupyter notebook

Dyllan2000alfa committed:
Fixed some things I forgot to change from base

Dyllan2000alfa committed:
Added readme and changelog

Dyllan2000alfa committed:
Merge branch 'truecharts:master' into jupyter

Crow-Control commented on charts/incubator/jupyter/values.yaml:
```diff
@@ -0,0 +1,52 @@
+  repository: jupyter/all-spark-notebook
+  tag: latest
+persistence:
+data:
```
```suggestion
  data:
```

Commit:
Update charts/incubator/jupyter/values.yaml

Crow-Control commented on charts/incubator/jupyter/values.yaml:
```diff
@@ -0,0 +1,52 @@
+  runAsUser: 568
+securityContext:
+  readOnlyRootFilesystem: true
+  runAsNonRoot: true
```
```suggestion
```

Commit:
Update charts/incubator/jupyter/values.yaml

stavros-k committed:
remove autogenerated content

stavros-k committed:
lint

stavros-k committed:
fix persistence and service

stavros-k commented:
It might be good to take into consideration this features.
https://jupyter-docker-stacks.readthedocs.io/en/latest/using/common.html#docker-options

Dyllan2000alfa commented:
Okay I had planed to at some point, was just trying to get basic functionality in the little time I had yesterday. I'll probably try and add most of the features once my classes are over for the day.

Dyllan2000alfa commented:
> Okay I had planed to at some point, was just trying to get basic functionality in the little time I had yesterday. I'll probably try and add most of the features once my classes are over for the day.

Ignore me closing and opening the PR I just woke up and clicked the wrong button trying to discard a comment.

stavros-k commented:
> Okay I had planed to at some point, was just trying to get basic functionality in the little time I had yesterday. I'll probably try and add most of the features once my classes are over for the day.

No rush! I can chime in if you want to, probably tomorrow

stavros-k committed:
Merge branch 'master' into jupyter

Dyllan2000alfa committed:
Merge branch 'master' into jupyter

Crow-Control commented:
Really don't worry, take your time on draft PR's :)

Dyllan2000alfa committed:
Merge branch 'master' into jupyter

Dyllan2000alfa committed:
Add some config options and try to fix permissions

I added some of the config options supported by the container. I also tried to fix permissions by setting UID and GID, but this doesn't seem to have worked

Dyllan2000alfa committed:
Changed container to run as root

I changed the pod security so that it would run as root in order for it to deploy with proper folder permissions.  From my understanding while the pod is running as root the actual jupyter process is running as user 568 since I set the uid and gid to 568.

Dyllan2000alfa committed:
Merge branch 'master' into jupyter

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/truecharts/charts?pullRequest=3616) <br/>All committers have signed the CLA.

stavros-k committed:
update IDs

stavros-k committed:
some improvements

stavros-k committed:
Merge branch 'master' into jupyter

Signed-off-by: Stavros Kois <47820033+stavros-k@users.noreply.github.com>

stavros-k committed:
typo

stavros-k committed:
update image repo

stavros-k committed:
update questions

stavros-k committed:
this is not needed, as users can select auto perms on persistance section

stavros-k commented on charts/incubator/jupyter/Chart.yaml:
```diff
@@ -0,0 +1,29 @@
+dependencies:
+  - name: common
+    repository: https://library-charts.truecharts.org
+    version: 10.5.4
```
```suggestion
    version: 10.5.7
```

stavros-k commented on charts/incubator/jupyter/questions.yaml:
```diff
@@ -0,0 +1,170 @@
+                              type: int
+                              default: 10300
+                              required: true
+# Include{advancedPortTCP}
```
```suggestion
# Include{advancedPortHTTP}
```

stavros-k committed:
Update charts/incubator/jupyter/questions.yaml

Signed-off-by: Stavros Kois <47820033+stavros-k@users.noreply.github.com>

stavros-k committed:
Update charts/incubator/jupyter/Chart.yaml

Signed-off-by: Stavros Kois <47820033+stavros-k@users.noreply.github.com>

stavros-k committed:
Merge branch 'master' into jupyter

stavros-k commented:
@all-contributors please add @Dyllan2000alfa  for code

truecharts-admin commented:
This PR is locked to prevent necro-posting on closed PRs. Please create a issue or contact staff on discord if you want to further discuss this
